src/runtime/providers/math.ts

```diff
@@ -63,6 +68,11 @@ function degreesToRadians(degrees:number){
 }
 
 class Add extends TotalFunctionConstraint {
+  resolveProposal(proposal, prefix) {
```

Comment:
A fair number of languages have explicit concatenation operators instead of overloading + for strings. The main argument against doing so is that it doesn't make a lot of sense without implicit conversions which is something we want to stay far away from. E.g. "foo" + 1 implicit converts 1 to a string in JS. I think we should do more than just fail silently in the case where you try to add strings, but we need to think a bit about what that warning system would be.